tests/ArgentPonyWarcraftClient.Extensions.DependencyInjection.Tests/ServiceCollectionExtensionsTests/When_Adding_Warcraft_Clients.cs

```diff
@@ -0,0 +1,107 @@
+using System;
+using System.Net.Http.Headers;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace ArgentPonyWarcraftClient.Extensions.DependencyInjection.Tests.ServiceCollectionExtensionsTests
+{
+    public class When_Adding_Warcraft_Clients
+    {
+        private readonly IServiceCollection _services;
+
+        public When_Adding_Warcraft_Clients()
+        {
+            _services = new ServiceCollection();
+
+        }
+
+        [Fact]
+        public void If_Client_Id_Is_Null_Then_ArgumentNullException_Is_Thrown()
+        {
+            var exception = Assert.Throws<ArgumentNullException>(() => 
+                _services.AddWarcraftClients(null, "fake-client-secret")
+            );
+
+            Assert.Equal("clientId", exception.ParamName);
+        }
+
+        [Fact]
+        public void If_Client_Secret_Is_Null_Then_ArgumentNullException_Is_Thrown()
+        {
+            var exception = Assert.Throws<ArgumentNullException>(() => 
+                _services.AddWarcraftClients("fake-client-id", null)
+            );
+
+            Assert.Equal("clientSecret", exception.ParamName);
+        }
+
+        [Theory]
+        [ClassData(typeof(WarcraftClientInterfaceData))]
```

Comment:
I like the use of the `ClassData` here.  Very clean!

Reply:
Thanks! I like trying to make unit tests easy to read and minimize duplication. This definitely felt like the easiest way to share the same test data between 2 different test classes 😄